### ISSUE TITLE
Intelligent backpace unindents to column 1 if preceeded by blank line

### ISSUE THREAD

**Issue by rlabbe** [spyder-ide/spyder#10146]
### What steps will reproduce the problem?

With intelligent backspace enabled, put the following in an empty editor file:

    def foo(x):
        for i in range(x):
            print(i)

With the cursor at the end of the print line, press return once, then backspace. The cursor correctly lines up under 'for' on column 4. This is correct behavior.

Now, put the cursor at the end of the print line, press return twice, and then backspace. Instead of lining up under 'for', the cursor goes all the way to column 1.

In 3.x the editor correctly ignores the blank line and deduces the correct indent level when you press backspace. 4.0.0b4 always goes back to column 1 if the previous line is blank.

## Versions

* Spyder version: 4.0.0b4 
* Python version: 3.7.3
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 7

### Dependencies

```
pygments >=2.0    :  2.4.2 (OK)
qdarkstyle >=2.6.4:  2.7 (OK)
sphinx >=0.6.6    :  2.1.2 (OK)
pyls >=0.27.0     :  0.28.2 (OK)
nbconvert >=4.0   :  5.5.0 (OK)
pandas >=0.13.1   :  0.25.1 (OK)
numpy >=1.7       :  1.16.4 (OK)
sympy >=0.7.3     :  1.4 (OK)
cython >=0.21     :  0.29.13 (OK)
qtconsole >=4.5.2 :  4.5.4 (OK)
IPython >=4.0     :  7.8.0 (OK)
matplotlib >=2.0.0:  3.1.1 (OK)
pylint >=0.25     :  2.3.1 (OK)
```

**Comment by ccordoba12** [comment]
This probably has to do with the automatic removal of blank lines added to Spyder 4, and I agree it's annoying.

@impact27, could you take a look at this one?

**Comment by bcolsen** [comment]
Is it possible to disable this? Spyder still removes the spaces it adds from automatic indents even when the "Automatically strip trailing spaces from changed lines" is disabled. 

I think there should be no whitespace removal or pep8 checking by default. This is only a code style warning not a coding error. Programming is already strict enough. Too strict for most people :-)

**Comment by impact27** [comment]
> I think there should be no whitespace removal

Do you mean not at all or should spyder remove the automatic indents? I think those should still be removed at least.

> or pep8 checking

I might agree with this one, but enabling the pep8 pushes the user to write better code. Should it be easier to activate / desactivate the linting?


> This probably has to do with the automatic removal of blank lines added to Spyder 4, and I agree it's annoying.

It is possible, but then what should be the algorithm? how far up should the intelligent backspace look? We can just add logic like: `if line.text() == "": line = line.previous()` to allow for one blank line. This would still happen for three returns but it is probably ok.

**Comment by ccordoba12** [comment]
> Do you mean not at all or should spyder remove the automatic indents? I think those should still be removed at least.

I agree with @bcolsen here. Removing all kinds of trailing spaces (even those associated with automatic indents) must be associated with turning on "Automatically strip trailing spaces from changed lines". If that option is off, then no removal should performed at all. Otherwise things are very confusing, as @bcolsen clearly noted above.

@impact27, could you take care of that? 

> I might agree with this one, but enabling the pep8 pushes the user to write better code. Should it be easier to activate / desactivate the linting?

I think there's no easier way than simply going to the Source menu and activating "Show code style warnings". So you're just two clicks away of pep8 checks.

> It is possible, but then what should be the algorithm? how far up should the intelligent backspace look? We can just add logic like: if line.text() == "": line = line.previous() to allow for one blank line. This would still happen for three returns but it is probably ok.

I agree with this. Could you also implement it?

**Comment by remisalmon** [comment]
This seems to have been fixed on Spyder 4.1.0 (can't reproduce the original issue)

**Comment by ccordoba12** [comment]
Right, this was fixed already. Thanks for noticing it.
